packages/apputils/src/widgettracker.ts

```diff
@@ -208,6 +210,12 @@ export class WidgetTracker<T extends Widget = Widget>
    * Add a new widget to the tracker.
    *
    * @param widget - The widget being added.
+   *
+   * #### Notes
+   * The widget passed into the tracker is added synchronously; its existence in
+   * the tracker can be checked with the `has()` method. The promise this method
+   * returns resolves after the widget has been added and saved to an underlying
+   * restoration connector, if one is available.
    */
   async add(widget: T): Promise<void> {
     this._focusTracker.add(widget);
```

Comment:
Do the initial synchronous operations in an `async` function happen synchronously the way they do in a non-`async` function? I'm unsure as to whether this `await this._pool.add(widget)` functionality is still identical to the previous API.

Reply:
Yes, this is identical. `await foo()` means "call `foo()` synchronously and get back a promise, synchronously" ... which means you can always ignore the promise if you don't care about its settlement.

Reply:
`this._pool.current = widget;` This line happens asynchronously, though.

Do you think it should be synchronous? I can place up above `await this._pool.add(widget);`

Reply:
My confusion comes from the fact that you can `await` a synchronous function, which pushes it onto the async stack, and it was unclear to me on which end that function evaluation happens.

If that is the case, then this looks good to me.

Reply:
Well, if you set `this._pool.current = widget` after the async part, then the signal emission is delayed until somewhat after you can safely start checking the tracker, right? I don't think that's necessarily a problem, but it is a bit subtle.

Reply:
Yeah I think this behavior is sound ... when `added` emits, `currentWidget` is correct and before then it's "still adding" but you can "do stuff with it".